tensorboard/webapp/runs/store/runs_reducers.ts

```diff
@@ -98,26 +99,38 @@ const dataReducer: ActionReducer<RunsDataState, Action> = createReducer(
       };
     }
 
-    const eidsBasedKey = serializeExperimentIds(action.experimentIds);
-    if (!nextSelectionState.has(eidsBasedKey)) {
-      const selectionMap = new Map<string, boolean>();
-      const runSelected =
-        action.runsForAllExperiments.length <=
-        MAX_NUM_RUNS_TO_ENABLE_BY_DEFAULT;
-      for (const run of action.runsForAllExperiments) {
-        selectionMap.set(run.id, runSelected);
+    /**
+     * Auto-select new runs based on the saved preference, or based on the
+     * number of runs.
+     *
+     * Known caveat: subsequent reloads after the first may result
+     * in selecting a large number of runs automatically, incuring a
+     * performance cost.
```

Comment:
Q: would it make sense to have some upper limit like 500 (random number off of my head)? 
OTOH, perhaps these rules are too random and may confuse users more than help them.

Reply:
$0.02: I think raising the limit significantly to N=100 or N=500 would probably make sense (especially for time series where we have the different query pattern, and the intersection observer, and the gpu line chart).

If we do keep a limit at all, it might be nice to make the behavior "select up to the limit, then activate a toast that explains not all runs were selected" rather than "past the limit, select 0 runs" like we have now.  That would also work a little better with the "new runs arrive" case IMO since we could just activate the toast again.  So e.g.

Initial page load:
> You have a lot of runs! Selected first N for display. [Dismiss]

Manual or auto-reload which loads new runs when N are already selected:
> New runs loaded, but not auto-selected since N are already selected. [Dismiss]